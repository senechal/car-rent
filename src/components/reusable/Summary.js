import PropTypes from 'prop-types'
import { RightOutlined } from '@ant-design/icons';
import Base from './Base';

const Summary = (props) => <Base defaultCls="rac-summary" {...props} />;
const Section = ({title, description, ...props}) => {
    return (
        <Base defaultCls="rac-summary-section" {...props}>
            <div className="rac-summary-section-detail">
                <div className="rac-summary-section-title">{title}</div>
                <div className="rac-summary-section-description">{description}</div>
            </div>
        </Base>
    );
};

const Separator = () => <RightOutlined className="rac-summary-separator" />

Section.propTypes = {
    title: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
};

Summary.Section = Section;
Summary.Separator = Separator;

export default Summary;
