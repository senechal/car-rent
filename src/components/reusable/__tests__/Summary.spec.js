import { render } from '@testing-library/react';
import Summary from '../Summary';

const { Section, Separator }  = Summary

describe('Testing Summary Components', () => {

  describe('testing Summary', () => {
    it('should render correctly', () => {
      const { container } = render(<Summary />);
      expect(container).toMatchSnapshot();
    });
  });
  describe('testing Separator', () => {
    it('should render correctly', () => {
      const { container } = render(<Separator />);
      expect(container).toMatchSnapshot();
    });
  });
  describe('testing Section', () => {
    it('should render correctly', () => {
      const { container } = render(<Section title="Title" description="description" />);
      expect(container).toMatchSnapshot();
    });
  });
});