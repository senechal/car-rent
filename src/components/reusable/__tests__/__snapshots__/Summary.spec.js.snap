// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Summary Components testing Section should render correctly 1`] = `
<div>
  <div
    class="rac-summary-section"
  >
    <div
      class="rac-summary-section-detail"
    >
      <div
        class="rac-summary-section-title"
      >
        Title
      </div>
      <div
        class="rac-summary-section-description"
      >
        description
      </div>
    </div>
  </div>
</div>
`;

exports[`Testing Summary Components testing Separator should render correctly 1`] = `
<div>
  <span
    aria-label="right"
    class="anticon anticon-right rac-summary-separator"
    role="img"
  >
    <svg
      aria-hidden="true"
      data-icon="right"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
      />
    </svg>
  </span>
</div>
`;

exports[`Testing Summary Components testing Summary should render correctly 1`] = `
<div>
  <div
    class="rac-summary"
  />
</div>
`;
